ml-agents/mlagents/trainers/stats.py

```diff
@@ -330,5 +335,6 @@ def get_stats_summaries(self, key: str) -> StatsSummary:
             std=np.std(stat_values),
             num=len(stat_values),
             sum=np.sum(stat_values),
+            full_dist=stat_values,
```

Comment:
This might be out of the scope of this PR, but I wonder if since we're feeding the entire list of `stat_values`, we can just compute `std`, `num`, `sum`, and `mean` as `@property` values inside the `StatsSummary`? e.g.
```
class StatssSummary(NamedTuple):
    full_dist: List[float]
    @property
    def mean(self):
        return mean(self.full_dist)
```

Reply:
This makes a lot of sense to me. The way the tests are all currently written involves a lot of hand-coded mean/std values though. I think it might necessitate a slightly bigger refactor to rework the tests to deal with the underlying values rather than their summary stats.

Reply:
I went ahead and made this change. I think it cleans up things nicely.